src/monster.cpp

```diff
@@ -2633,17 +2633,19 @@ void monster::process_effects()
 
     Character &player_character = get_player_character();
     //If this monster has the ability to heal in combat, do it now.
-    const int healed_amount = heal( type->regenerates );
-    if( healed_amount > 0 && one_in( 2 ) ) {
-        std::string healing_format_string;
-        if( healed_amount >= 50 ) {
-            healing_format_string = _( "The %s is visibly regenerating!" );
-        } else if( healed_amount >= 10 ) {
-            healing_format_string = _( "The %s seems a little healthier." );
-        } else {
-            healing_format_string = _( "The %s is healing slowly." );
+    if( type->regenerates && !has_effect( "onfire" ) ) {
```

Comment:
This should check an effect flag, not a hardcoded effect id.

Reply:
> This should check an effect flag, not a hardcoded effect id.

```suggestion
    if( type->regenerates && !has_effect( effect_onfire ) ) {
```
This way?

Reply:
No, that's still a hardcoded effect id. This should be a property that can be applied to any effect, and is not hardcoded to a single specific effect.